### PR TITLE
Feature-detection of chrome.sockets

### PR DESCRIPTION
**Pull request by lidel**
> Part of an effort to run embedded js-ipfs in Brave :lion: https://github.com/ipfs-shipyard/ipfs-companion/issues/664  

## TL;DR

This PR adds runtime dectection of `chrome.sockets` APIs and a build target for testing it with [Brave Nightly](https://brave.com/download-nightly/): `yarn dev-build bundle:brave`

## What is new?

This PR adds new tests and should not modify existing functionality unless user is running Brave Nightly. 

External node type remains the default, but Brave users can opt-in to "_Embedded + chrome.sockets_" node type on the Preferences screen, which will enable Embedded HTTP Gateway.

> 2019-04-16-220411_1133x593_scrot

## Details

This PR includes cherry-picked js-ipfs improvements:

- streaming compressed payload (https://github.com/ipfs/js-ipfs/pull/1989)
  - Content-type sniffing is now done over a meaningful amount of bytes instead of arbitrary number.
- decoupling HTTP Servers from cli/commands/daemon (https://github.com/ipfs/js-ipfs/pull/1950)

Other
  - Brave users will see a new node type: `embedded:chromesockets` (https://github.com/ipfs-shipyard/ipfs-companion/commit/5c0b495b5d41c80352ced2678b4b7a8acfaa0df9)
  - cleanup boot on clean install (https://github.com/ipfs-shipyard/ipfs-companion/commit/2b636360944397ab257f9990cdc390f14931ce50)
  - new runtime-checks: `isBrave`, `hasChromeSocketsForTcp` 
  - peer discovery over a ws-star to  improve things until we have TCP transport, MulticastDNS and DHT

### CODE REVIEW

**Comment by lidel** [comment]
_(Merging this in fast-track mode as we want to ship this to Beta channel this week to unblock development on Brave side)_
